frontend/src/components/SyncEditor/process.ts

```diff
@@ -114,71 +124,83 @@ const process = (
+    cachedSecrets?.forEach(({ path, value }) => {
         if (has(parsed, path)) {
             set(parsed, path, value)
         }
     })
-    let changeWithSecrets = { yaml, mappings, parsed, resources, hiddenSecretsValues }
 
-    // hide and remember secret values
-    const protectedRanges: any[] = []
-    const hideSecrets = secrets && !isEmpty(parsed) && !isEmpty(secrets)
-    if (hideSecrets) {
-        changeWithSecrets = {
-            yaml,
-            mappings: cloneDeep(mappings),
-            parsed: cloneDeep(parsed),
-            resources: cloneDeep(resources),
-            hiddenSecretsValues,
+    // restore omitted filter values
+    cacheFiltered?.forEach(({ path, value }) => {
+        if (has(parsed, path)) {
+            set(parsed, path, value)
         }
+    })
 
-        // expand wildcards in declared secret paths
-        const allSecrets = getAllPaths(secrets, mappings, parsed)
+    const unredactedChange = {
+        yaml,
+        mappings: cloneDeep(mappings),
+        parsed: cloneDeep(parsed),
+        resources: cloneDeep(resources),
+        hiddenSecretsValues,
+        hiddenFilteredValues,
+    }
 
+    // hide and remember secret values
+    const protectedRanges: any[] = []
+    if (!isEmpty(parsed)) {
         // stuff secrets with '*******'
-        allSecrets.forEach(({ path }) => {
-            const value = get(parsed, path) as unknown as string
-            if (value && typeof value === 'string') {
-                hiddenSecretsValues.push({ path: path, value })
-                set(parsed, path, `${'*'.repeat(Math.min(20, value.replace(/\n$/, '').length))}`)
-            }
-        })
+        let allSecrets: { path: string | any[]; isRange: boolean }[] = []
+        if (secrets && !isEmpty(secrets)) {
+            allSecrets = getAllPaths(secrets, mappings, parsed)
 
-        // filter kube resources
-        if (filterResources) {
-            resources = filterKubeResources(resources)
+            allSecrets.forEach(({ path }) => {
+                const value = get(parsed, path) as unknown as string
+                if (value && typeof value === 'string') {
+                    hiddenSecretsValues.push({ path: path, value })
+                    set(parsed, path, `${'*'.repeat(Math.min(20, value.replace(/\n$/, '').length))}`)
+                }
+            })
         }
 
-        // create yaml with '****'
+        // stuff filtered with '-filtered-'
+        let allFiltered: { path: string | any[]; isRange: boolean }[] = []
+        if (filters && !isEmpty(filters)) {
+            allFiltered = getAllPaths(filters, mappings, parsed)
+            allFiltered.forEach(({ path }) => {
+                const value = get(parsed, path) as unknown as string
+                if (value && typeof value === 'object') {
+                    hiddenFilteredValues.push({ path: path, value })
+                    set(parsed, path, '-click unfilter YAML to show-')
```

Comment:
Please:
- use uppercase U in Unfilter to match tooltip on filter button
- externalize your strings